cli/cmd/repair.go

```diff
@@ -43,6 +51,27 @@ linkerd upgrade.`,
 				return errors.New("secret/linkerd-config-overrides already exists. If you need to regenerate this resource, please delete it before proceeding")
 			}
 
+			// Check if the CLI version matches with that of the server
+			clientVersion := version.Version
+			var serverVersion string
+			publicClient, err := public.NewExternalClient(cmd.Context(), controlPlaneNamespace, k8sAPI)
+			if err != nil {
+				return err
+			}
+			serverVersion, err = healthcheck.GetServerVersion(cmd.Context(), publicClient)
+			if err != nil {
+				return err
```

Comment:
Given these are not errors related to the commands flags, can you echo the error and do `os.Exit(1)` like we started doing with the other CLI commands? (and while you're at it do the same for the other returns above and below :wink: )

Reply:
Updated and made it simpler by moving it into a separate function instead